diktat-common/src/main/kotlin/org/cqfn/diktat/common/config/rules/RulesConfigReader.kt

```diff
@@ -122,7 +123,9 @@ data class CommonConfiguration(private val configuration: Map<String, String>?)
      */
     val kotlinVersion: KotlinVersion by lazy {
         configuration?.get("kotlinVersion")?.kotlinVersion() ?: run {
```

Comment:
Did you figure out why it was possible? Body of lazy should have been executed only once, even with parallel execution. Counter fixes this behavior, of course, but it's always better to find a root cause. Did you try to reproduce it in maven, for example?

Reply:
This was due to our `List<RulesConfig>.getCommonConfiguration()`  function. To fix, I can make an atomic counter (as already done) or make a lazy extension property.

Reply:
I suggest to use lazy extension property, I think using a function there was by mistake.

Reply:
Unfortunately, a lazy property requires storing state, and there is no way to add state to an existing object from an extension property.